src/overlay005/fieldmap.c

```diff
@@ -0,0 +1,1002 @@
+#include "overlay005/ov5_021D5878.h"
+#include "overlay005/ov5_021D5B40.h"
+#include "overlay005/ov5_021D5BC0.h"
+#include "overlay005/ov5_021D5CB0.h"
+#include "overlay005/ov5_021D5EB8.h"
+#include "overlay005/ov5_021DD6FC.h"
+#include "overlay005/ov5_021DF440.h"
+#include "overlay005/ov5_021E15F4.h"
+#include "overlay005/ov5_021E1B08.h"
+#include "overlay005/ov5_021E779C.h"
+#include "overlay005/ov5_021EA714.h"
+#include "overlay005/ov5_021ECC20.h"
+#include "overlay005/ov5_021ECE40.h"
+#include "overlay005/ov5_021EE75C.h"
+#include "overlay005/ov5_021EF250.h"
+#include "overlay005/ov5_021EF3A8.h"
+#include "overlay005/ov5_021EF4BC.h"
+#include "overlay005/ov5_021EF75C.h"
+#include "overlay005/ov5_021EFB0C.h"
+#include "overlay005/ov5_021F0824.h"
+#include "overlay005/ov5_021F10E8.h"
+#include "overlay009/ov9_02249960.h"
+
+FS_EXTERN_OVERLAY(overlay6);
+FS_EXTERN_OVERLAY(overlay7);
+FS_EXTERN_OVERLAY(overlay8);
+FS_EXTERN_OVERLAY(overlay9);
+
+struct UnkStruct_ov5_021D1A68_t {
+    u16 unk_00;
+    int unk_02[24];
+};
+
+static void ov5_021D1444(BGL * param0);
+static void ov5_021D1524(BGL * param0);
+static void ov5_021D154C(void);
+static void ov5_021D1570(void);
+static void ov5_021D1578(UnkStruct_ov5_021D5894 * param0);
+static void ov5_021D15F4(FieldSystem * param0);
+static void ov5_021D173C(FieldSystem * param0);
+static void ov5_021D1414(void);
+static void ov5_021D15B4(void);
+static void ov5_021D15E8(void);
+static void ov5_021D1790(FieldSystem * param0);
+static void ov5_021D17EC(FieldSystem * param0);
+static void ov5_021D1878(FieldSystem * param0);
+static void ov5_021D1968(FieldSystem * param0);
+static int FieldMap_Init(OverlayManager * param0, int * param1);
+static int FieldMap_Main(OverlayManager * param0, int * param1);
+static int FieldMap_Exit(OverlayManager * param0, int * param1);
+static BOOL FieldMap_ChangeZone(FieldSystem * param0);
+static void ov5_021D134C(FieldSystem * param0, u8 param1);
+static BOOL ov5_021D119C(FieldSystem * param0);
+static void ov5_021D0D80(void * param0);
+static void ov5_021D13B4(FieldSystem * param0);
+static int ov5_021D1178(FieldSystem * param0);
+static BOOL FieldMap_InDistortionWorld(FieldSystem * param0);
+static UnkStruct_ov5_021D1A68 * ov5_021D1A14(int param0, int param1);
+static const int * ov5_021D1A68(const UnkStruct_ov5_021D1A68 * param0);
+static const int ov5_021D1A6C(const UnkStruct_ov5_021D1A68 * param0);
+static void ov5_021D1A70(UnkStruct_ov5_021D1A68 * param0);
+
+
+static inline void inline_ov5_021D0D80 (FieldSystem * fieldSystem)
+{
+    UnkStruct_ov5_021ED0A4 * v0 = sub_0206285C(fieldSystem->mapObjMan);
+    UnkStruct_02020C44 * v1 = ov5_021EDC8C(v0);
+
+    sub_02020D68(v1);
+}
+
+static void ov5_021D0D80 (void * param0)
+{
+    FieldSystem * fieldSystem = param0;
+
+    sub_0201C2B8(fieldSystem->unk_08);
+    sub_0201DCAC();
+    sub_0200A858();
+
+    inline_ov5_021D0D80(fieldSystem);
+}
+
+static int FieldMap_Init (OverlayManager * param0, int * param1)
```

Comment:
The `Init`, `Main`, and `Exit` functions should return `BOOL` to align with the `OverlayFunc` typedef.